ui/lib/apps/TopSQL/utils/useTimeWindowSize.ts

```diff
@@ -8,10 +9,11 @@ export const createUseTimeWindowSize = (barWidth: number) => {
     const [timeWindowSize, setTimeWindowSize] = useState<number>(0)
     const computeTimeWindowSize = (
       screenWidth: number,
-      totalTimeRange: number
+      [min, max]: TimeRangeValue
     ) => {
-      const windowSize = (barWidth * totalTimeRange) / screenWidth
-      setTimeWindowSize(Math.ceil(windowSize))
+      const windowSize = Math.ceil((barWidth * (max - min)) / screenWidth)
+      setTimeWindowSize(windowSize)
+      return windowSize
```

Comment:
I have extracted a `useChartHandle.calcIntervalSec`, which may be useful for you. BTW the time window size should be a ref instead of a state, because when it is changed nothing should be re-rendered. The time window size is only something used at the request moment I believe.

Reply:
There's a feature that can resize the bar series width and adjust the window size depending on the container width.